### PR TITLE
View/Angle Offsetting

### PR DESCRIPTION
**Pull request by MajorCooke**
Prototype for player view offsetting.

This includes the ability to make offsetted views via position and angle/yaw/pitch without affecting the actual three angles and position of the player actor.

### CODE REVIEW

File: src/playsim/d_player.h [hunk 1]
```diff
@@ -88,6 +88,11 @@ void PlayIdle(AActor *player);
 enum
 {
 	PPF_NOTHRUSTWHENINVUL = 1,	// Attacks do not thrust the player if they are invulnerable.
+	PFF_VIEWABSOFFSET =		1 << 3,
+	PFF_VIEWABSPOS =		1 << 4,
+	PFF_VIEWABSANGLE =		1 << 5,
+	PFF_VIEWABSPITCH =		1 << 6,
+	PFF_VIEWABSROLL =		1 << 7,
```
Comment (madame-rachelle) [inline src/playsim/d_player.h]: There is an inconsistency here. The supplied flag is PPF_*, the new addition here is PFF_*

Comment (MajorCooke) [reply src/playsim/d_player.h]: Thanks for pointing that out! I completely overlooked that. And it helped me fix another small bug with the roll.

**Comment by coltongit** [comment]
@MajorCooke, is this in a state where QZDoom could take this on?

**Comment by MajorCooke** [comment]
No. I've chosen part 1 instead for now. https://github.com/coelckers/gzdoom/pull/927

**Comment by madame-rachelle** [comment]
What is the state of this PR? It has a merge conflict which looks like it might be pretty easy to resolve, but not much recent discussion on this.

**Comment by MajorCooke** [comment]
First I'd like to get [this PR](https://github.com/coelckers/gzdoom/pull/1002) merged, because in the event I cannot get this working appropriately with portals, it doesn't hold up the first section since i know that behaves properly.

The problem with the positional offsetting of the view is portals. While it's mainly just a visual issue and doesn't affect anything else, it definitely is quite weird to deal with.

**Comment by MajorCooke** [comment]
Since #1002 was merged, I'm closing this and will rework the other half in another PR.

**Pull request closed without merging**